Add guides to monitor Tyk (#2096)
TykTechnologies/tyk-docs: tyk-docs/content/planning-for-production/monitoring/tyk-component-monitoring.md

---
date: 2017-03-24T10:10:41Z
title: Tyk Components
tags: ["Monitoring", "Observability", "SLO", "infrastructure"]
description: "How to set up monitoring and observability of your API kingdom"
weight: 1
menu:
  main:
    parent: "Monitoring"
url: "/planning-for-production/monitoring/tyk-components"
---

# Monitoring Tyk Components

A common question that gets asked is how to monitor the Tyk components.

## Tyk Gateway

> The Gateway & Redis are the only components that will have a high on demand performance requirement, that scales with your API traffic.  

Reviewer: ```suggestion
The Gateway & Redis are the only components that will have a high on-demand performance requirement, which needs to scale with your API traffic.
```

> ### CPU Utilisation

Reviewer: ```suggestion
### Tyk Gateway CPU Utilisation
```

Performance benchmarks on how Tyk performs across different CPU architectures, environments and sizes [here](https://tyk.io/performance-benchmarks/).

A healthy and performant Tyk Gateway should have a CPU utilisation of under 60%. If the average CPU utilisation is above 60%, then we recommend you scale your Tyk Gateway services. A higher figure than 60% introduces risk, because if one Gateway fails, the traffic spillover to healthy nodes might be overwhelming and result in cascading failure.

### Liveness Health Check

Health checks are extremely important in determining the status of our Tyk Components. Depending on your setup and configuration, the statuses of the following components will be returned: **Gateway**, **Dashboard**, **Redis** and **RPC**. 

The health check endpoint is an expensive operation and can throttle throughput so it's important to find an optimal plan if you want to take advantage of this endpoint. The endpoint refreshes every 10 seconds and does not return statuses on the primary database (MongoDB or PostgreSQL) and the Tyk Pump component.

> ```

Reviewer: ```suggestion
```console
```
